nomos-da/network/core/src/swarm/validator.rs

```diff
@@ -83,33 +119,32 @@ where
         }
     }
 
-    pub async fn step(&mut self) {
-        tokio::select! {
-            Some(event) = self.swarm.next() => {
+    pub async fn run(mut self) {
+        loop {
+            if let Some(event) = self.swarm.next().await {
                 match event {
                     SwarmEvent::Behaviour(behaviour_event) => {
                         self.handle_behaviour_event(behaviour_event).await;
-                    },
-                    SwarmEvent::ConnectionEstablished{ .. } => {}
-                    SwarmEvent::ConnectionClosed{ .. } => {}
-                    SwarmEvent::IncomingConnection{ .. } => {}
-                    SwarmEvent::IncomingConnectionError{ .. } => {}
-                    SwarmEvent::OutgoingConnectionError{ .. } => {}
-                    SwarmEvent::NewListenAddr{ .. } => {}
-                    SwarmEvent::ExpiredListenAddr{ .. } => {}
-                    SwarmEvent::ListenerClosed{ .. } => {}
-                    SwarmEvent::ListenerError{ .. } => {}
-                    SwarmEvent::Dialing{ .. } => {}
-                    SwarmEvent::NewExternalAddrCandidate{ .. } => {}
-                    SwarmEvent::ExternalAddrConfirmed{ .. } => {}
-                    SwarmEvent::ExternalAddrExpired{ .. } => {}
-                    SwarmEvent::NewExternalAddrOfPeer{ .. } => {}
+                    }
+                    SwarmEvent::ConnectionEstablished { .. } => {}
```

Comment:
What do you think of adding already (at least debug) logs for each of these events? It might be helpful for debugging.

Reply:
Yeah, I could add one debug at the top of this method, or instrument it